src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Telemetry/VSTelemetryServiceUtility.cs

```diff
@@ -66,33 +67,45 @@ public static class VSTelemetryServiceUtility
 
         public static async Task<ProjectTelemetryEvent> GetProjectTelemetryEventAsync(NuGetProject nuGetProject)
         {
-            // Get the project details.
-            string projectUniqueName = nuGetProject.GetMetadata<string>(NuGetProjectMetadataKeys.UniqueName);
+            if (nuGetProject == null)
+            {
+                throw new ArgumentNullException(nameof(nuGetProject));
+            }
+            string projectUniqueName = "";
```

Comment:
nit: string.Empty;

Reply:
I thought `""` is faster than `string.Empty`; I need to investigate.

Reply:
It used to be (still is?) the other way around. `""` causes a memory allocation, and therefore maybe a GC, whereas `string.Empty` returns a static.  I'm not 100% sure, but I think the compiler now treats `""` as an alias for `string.Empty`.  But oldies like me keep parroting `string.Empty` because it definately did not happen in the past.

Reply:
Both are interned. So no there's definitely no perf differences.
Many prefer using string.Empty is more commonly than `""` because of how explicit it is. 

After all that's why my comment is a `nit`.

Reply:
Compiling this program in a sdk-style project targeting net472:

```
using System;

namespace str.empty
{
    class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine(string.Empty);
            Console.WriteLine("");
        }
    }
}
```

gives me the following IL code:

```
.method private hidebysig static void  Main(string[] args) cil managed
{
  .entrypoint
  // Code size       24 (0x18)
  .maxstack  8
  IL_0000:  nop
  IL_0001:  ldsfld     string [mscorlib]System.String::Empty
  IL_0006:  call       void [mscorlib]System.Console::WriteLine(string)
  IL_000b:  nop
  IL_000c:  ldstr      ""
  IL_0011:  call       void [mscorlib]System.Console::WriteLine(string)
  IL_0016:  nop
  IL_0017:  ret
} // end of method Program::Main
```

Reply:
Analyzing JIT asm code, both calls are identical:

```
; Desktop CLR v4.8.4261.00 on x86

str.empty.Program..ctor()
    L0000: push ebp
    L0001: mov ebp, esp
    L0003: push eax
    L0004: mov [ebp-0x4], ecx
    L0007: cmp dword [0x1e1a2f4c], 0x0
    L000e: jz L0015
    L0010: call 0x730bfd80
    L0015: mov ecx, [ebp-0x4]
    L0018: call System.Object..ctor()
    L001d: nop
    L001e: nop
    L001f: mov esp, ebp
    L0021: pop ebp
    L0022: ret

str.empty.Program.Main(System.String[])
    L0000: push ebp
    L0001: mov ebp, esp
    L0003: push eax
    L0004: mov [ebp-0x4], ecx
    L0007: cmp dword [0x1e1a2f4c], 0x0
    L000e: jz L0015
    L0010: call 0x730bfd80
    L0015: nop
    L0016: mov ecx, [0xe5d2340]
    L001c: call System.Console.WriteLine(System.String)
    L0021: nop
    L0022: mov ecx, [0xe5d2340]
    L0028: call System.Console.WriteLine(System.String)
    L002d: nop
    L002e: nop
    L002f: mov esp, ebp
    L0031: pop ebp
    L0032: ret
```

https://sharplab.io/#v2:EYLgdgpgLgZgHgGiiAhgZwLYICYgNQA+AAgEwCMAsAFDVgoYRoAOKAxhAARpQBOAdBAxMoAT2oBvah2kdSssgHYpMyVRnr5ANlkAWDgFkUASzAAKImQAMAbQC6HFDwDmaAJTKNHVZ88WAnOZWfACiQqKuANwePtL+pgBE8ZHR6gC+0elUqUA

Reply:
In conclusion, I will use `string.Empty` to favor more readable code.